src/core/annotation.js

```diff
@@ -646,6 +646,12 @@ class Annotation {
       this._streams.push(this.appearance);
     }
 
+    if (dict.has("PMD")) {
```

Comment:
Given that this should be limited to widget annotations (both in the PDF file from the bug and in general), can we move this check to the specific `WidgetAnnotation` class instead?

Moreover, previously we did https://github.com/mozilla/pdf.js/pull/13214/files#diff-a7e409abd3d3209d32d3adc8457c5e23feb4c06113a2e065b0c6a9322a3a3369L1155-L1162 to hide widget annotations, but I doubt barcodes will have a field value and the `data.hidden` field should already be set correctly if you change the flags like in this PR (which is probably also preferable over the `setFlags` call which would wipe the existing flags from an API perspective).

Reply:
`annotationFlags` is a property we expose for annotations via the API; see https://github.com/mozilla/pdf.js/blob/bb73d2a92240a5177268a652f400a98805f83682/test/unit/annotation_spec.js#L4100 for a test and https://github.com/mozilla/pdf.js/blob/bb73d2a92240a5177268a652f400a98805f83682/src/core/annotation.js#L666-L683 for the data we expose publicly for annotations. It's probably not ideal, but the idea was to consolidate and explicitly specify which properties we expose to the outside via the API and by definition have the rest remain private, which also made API changes more explicit.

In that sense we can't remove it without this becoming an API change, but we also shouldn't need to remove it I think because the way you change the flags here should be fine and automatically be reflected in `data.annotationFlags` and `data.hidden` if it's set before the `super` call.

The reason for moving it would be to only do this extra check for specific annotation types where this can occur, to avoid having to do this for all annotation types unconditionally. It's not a dealbreaker, but it would be nice to keep this as locally as possible.